### PR TITLE
Refactors area creation

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: Humans and drones now have a UI button that allows them to create new areas, in the same manner as blueprints. Unlike blueprints, this button does not allow you to edit existing areas.
add: You can now use mineral doors (or any other object that blocks atmos) to complete rooms with blueprints
/:cl:

Refactored area creation code into a general proc, tried cleaning up bits of it, gave mobs a button to let them build new areas so you can build your lavaland or space hut without begging the CE.

Permits are gone, they no longer serve any use.

TODO:

Give artificers and alien drones a way to call the area creation proc

### CODE REVIEW

**Comment by Iamgoofball** [comment]
blueprints even more useless

give CE improved speed tools 2k16

:+1:

**Comment by Alek2ander** [comment]
This is something that's used so infrequently that giving it a separate UI button doesn't seem very reasonable.

**Comment by xxalpha** [comment]
Agreed, make it a verb or an item.

**Comment by KorPhaeron** [comment]
http://i.imgur.com/VCQ67wD.png

It's a bit ugly right now with that sprite, but the little box fits very nicely up against the crafting button.

And yeah building might be too rare to justify a UI button, but building might become more common if anyone is able to do it without having to beg for a traitor objective.

File: code/game/objects/items/blueprints.dm [hunk 1]
```diff
@@ -179,8 +149,7 @@
 			return AREA_SPECIAL
 	return AREA_STATION
 
-
-/obj/item/areaeditor/proc/create_area()
+/proc/create_area(mob/living/usr)
```
Comment (RemieRichards) [inline code/game/objects/items/blueprints.dm]: don't name vars like this, ```usr``` is taken and this could cause issues.

Comment (KorPhaeron) [inline code/game/objects/items/blueprints.dm]: alright

**Comment by RemieRichards** [comment]
@xxalpha No verbs.

**Comment by PKPenguin321** [comment]
but this is a pretty textbook case where a verb would work though

**Comment by RemieRichards** [comment]
@PKPenguin321 except we don't want verbs anymore, the goal is to eventually have none of them (none visible atleast, hidden ones which only work for text are fine (mainly to appease MSO) so that we can kill off the verb tabs and have a better screen layout. We started this by banning object verbs, which were slow and clunky and bad, and now we've on to them all.

Can't do the above if somebody keeps adding more verbs. (Admin shit's fine cos Admin UI/Controls all need a redo)

**Comment by Shadowlight213** [comment]
This seems like a really dumb thing to make a ui button for.
It could be made something buildable in the autolathe or even just something you make with paper.

**Comment by TechnoAlchemisto** [comment]
Even though it's small it seems like a bad thing to use UI space on

**Comment by KorPhaeron** [comment]
http://i.imgur.com/OpCHaI2.png

I redid the crafting button to be a bit less ugly as well.

Also I don't know where else to put it but UI if verbs are out. Having an autolathe item wouldn't help mobs meant to build things like artificers or xenomorphs, nor would it help ash walkers/hermits/whatever.

**Comment by Shadowlight213** [comment]
@KorPhaeron as dumb as it may sound..
Make a "Create area" spell/ability for non humans?
Also ash walkers could maybe get like a flag or stake thing to "claim a hunting ground/base"
Something similar for hermits.

**Comment by Bawhoppen** [comment]
Make it so you have to have certain prerequisites to do it, because areas are supposed to essentially imply a roof, so make it so you need to have a wall or something nearby so you can't just make a random safe space in lavaland with nothing.

**Comment by RemieRichards** [comment]
@Bawhoppen that's literally how it works.
You have to have an enclosed space (with or w/o doors)

**Comment by Bawhoppen** [comment]
okay well thats sensible

**Pull request merged**